### PR TITLE
Query pack updates

### PR DESCRIPTION
**Pull request by jacob-ronstadt**
## Checklist for Pull Requests
Add RoleTypeCorrectlyUsed query to recommended suite
Update README for using qlpacks
update version

- [ ] Description is filled out.
- [ ] Only one query or related query group is in this pull request.
- [ ] The version number on changed queries has been increased via the `@version` comment in the file header.
- [ ] All unit tests have been run: ([Test README.md](src\drivers\test\README.md)).
- [ ] Commands `codeql database create` and `codeql database analyze` have completed successfully.
- [ ] A .qhelp file has been added for any new queries or updated if changes have been made to an existing query.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,14 +44,19 @@ For general use, use the `main` branch along with [version 2.15.4 of the CodeQL
...
 
-1. Install CodeQL Packages using `codeql pack install`
+1. Install CodeQL Packages 
+    
+    For WHCP_21H2 and WHCP_22H2 branches:
+ 
+    **NOTE** Visual Studio 17.8 broke compatibility with the older versions of CodeQL used in the WHCP_21H2 and WHCP_22H2 branches. CodeQL 2.15.4 has been validated for use with WHCP 21H2 and WHCP 22H2. If using Visual Studio version 17.8+, follow the steps for ALL OTHER BRANCHES, below. **Make sure to remove the CodeQL submodule if it exists in your repo.** CodeQL might try to use the queries in the submodule by default which will cause errors because of mismatched versions.
 
-   For WHCP BRANCHES: Skip this step.
+    The WHCP branches WHCP_21H2 and WHCP_22H2 use the CodeQL repository as a submodule instead of using CodeQL packs. When using these branches, make sure to update submodules.
```
Comment (NateD-MSFT) [inline README.md:53]: This paragraph is a bit confusing, because it provides a notice about using 17.8 and CodeQL packs, then immediately instructs people to use the "ALL OTHER BRANCHES" section, and then follows that telling users to update submodules.  I think we should reword this to make it more clear that if and only if you are on VS 17.7 or below should you use submodules for these branches, and for all other cases (17.8 or other branches) you should use the pack commands.

File: README.md [hunk 2]
```diff
@@ -64,14 +44,19 @@ For general use, use the `main` branch along with [version 2.15.4 of the CodeQL
...
+    The WHCP branches WHCP_21H2 and WHCP_22H2 use the CodeQL repository as a submodule instead of using CodeQL packs. When using these branches, make sure to update submodules. 
 
-   For MAIN AND DEVELOPMENT BRANCHES use:
+    For ALL OTHER BRANCHES run the following commands:
 
-   ```
-    D:\codeql-home\codeql>codeql pack install D:\codeql-home\Windows-Driver-Developer-Supplemental-Tools\src
+    ```
+    codeql pack download microsoft/windows-drivers
+    codeql pack install <package install location>\<package name>\<package version>\
```
Comment (NateD-MSFT) [inline README.md:59]: Why is the codeql pack install command left with to-be-filled arguments? What is the expected action the user should take here? (i.e. are they installing the dependencies for our pack?)

File: README.md [hunk 3]
```diff
@@ -89,11 +74,25 @@ For general use, use the `main` branch along with [version 2.15.4 of the CodeQL
     
     CodeQL's analysis output is provided in the form of a SARIF log file. For a human readable format, drop the SARIF file into [SARIF Viewer Website](https://microsoft.github.io/sarif-web-component/). (If there are violations, they will show up. If not, the page will not update.)
     ```
-    D:\codeql-home\codeql>codeql database analyze <path to database> --format=sarifv2.1.0 --output=<"path to output file".sarif> <path to query/suite to run>
+    codeql database analyze <path to database> microsoft/windows-drivers --format=sarifv2.1.0 --output=<"path to output file".sarif> 
+    ```
+    By default, the `microsoft/windows-drivers` CodeQL pack will run the windows_driver_mustfix query suite. However, specific versions, queries, or suites can be specified using the format `codeql database analyze <database> <scope>/<pack>@x.x.x:<path>`. For futher information, see the [CodeQL documentation](https://docs.github.com/en/code-security/codeql-cli/using-the-advanced-functionality-of-the-codeql-cli/publishing-and-using-codeql-packs#using-a-codeql-pack-to-analyze-a-codeql-database)
```
Comment (NateD-MSFT) [inline README.md:79]: This is less a documentation point and more a policy one - do you think it'd be better to have the pack run recommended by default, since that's a superset of must-fix? That way users get more queries running and we get more feedback.  We can include a note that the must-fix suite is also available.

File: README.md [hunk 4]
```diff
@@ -88,14 +96,27 @@ For general use, use the `main` branch along with [version 2.15.4 of the CodeQL
 1. Analyze your CodeQL database:
     
     CodeQL's analysis output is provided in the form of a SARIF log file. For a human readable format, drop the SARIF file into [SARIF Viewer Website](https://microsoft.github.io/sarif-web-component/). (If there are violations, they will show up. If not, the page will not update.)
+
+    CodeQL query suites are provided in the suites directory and contain the sets of all recommended and mustfix queries. The desired query suite file should be downloaded/copied locally.
```
Comment (NateD-MSFT) [inline README.md:100]: Is it correct that we are having people download the suites, or are we moving to having people download the bundle and then just need to run one CLI command per your convo with GitHub earlier?

Comment (jacob-ronstadt) [reply README.md:100]: They will still need to download the suites as I found out there are still issues with using the bundle

File: README.md [hunk 5]
```diff
@@ -146,3 +169,32 @@ trademarks or logos is subject to and must follow
 [Microsoft's Trademark & Brand Guidelines](https://www.microsoft.com/en-us/legal/intellectualproperty/trademarks/usage/general).
 Use of Microsoft trademarks or logos in modified versions of this project must not cause confusion or imply Microsoft sponsorship.
 Any use of third-party trademarks or logos are subject to those third-party's policies.
+
+
+## Special instructions for WHCP_21H2 and WHCP_22H2 using VS17.7 or below
+1. Install Codeql version as indicated in above steps.
```
[the remark this answers is not in the document]
Comment (jacob-ronstadt) [reply README.md:175]: These instructions would only apply when using the older versions of CodeQL. I'll update the readme to specify that.

File: README.md [hunk 6]
```diff
@@ -88,14 +96,27 @@ For general use, use the `main` branch along with [version 2.15.4 of the CodeQL
 1. Analyze your CodeQL database:
     
     CodeQL's analysis output is provided in the form of a SARIF log file. For a human readable format, drop the SARIF file into [SARIF Viewer Website](https://microsoft.github.io/sarif-web-component/). (If there are violations, they will show up. If not, the page will not update.)
+
+    CodeQL query suites are provided in the suites directory and contain the sets of all recommended and mustfix queries. The desired query suite file should be downloaded/copied locally.
+    
+    1. Create a local copy of the desired query suite file:
+       
+        * windows_all_mustfix.qls
```
[the remark this answers is not in the document]
Comment (jacob-ronstadt) [reply README.md:108]: noted

File: README.md [hunk 7]
```diff
@@ -146,3 +171,35 @@ trademarks or logos is subject to and must follow
 [Microsoft's Trademark & Brand Guidelines](https://www.microsoft.com/en-us/legal/intellectualproperty/trademarks/usage/general).
 Use of Microsoft trademarks or logos in modified versions of this project must not cause confusion or imply Microsoft sponsorship.
 Any use of third-party trademarks or logos are subject to those third-party's policies.
+
+
+## Special instructions for WHCP_21H2 and WHCP_22H2 using VS17.7 or below
+These instructions only apply when using both Visual Studio 17.7 or below along with CodeQL 2.6.3 or 2.4.6
+
+
+1. Install Codeql version as indicated in above steps.
```
Comment (NateD-MSFT) [inline README.md:180]:
```suggestion
1. Install CodeQL version as indicated in above steps.
```

**Review by NateD-MSFT — APPROVED** [review]
